sycl/doc/extensions/GroupAlgorithms/SYCL_INTEL_group_sort.asciidoc

```diff
@@ -0,0 +1,325 @@
+|===
+
+Example of custom Sorter:
+[source,c++]
+----
+template<typename Compare>
+class bubble_sort{
+public:
+  Compare comp;
+
+  template<typename Group, typename Ptr>
+  void operator()(Group g, Ptr begin, Ptr end){
+    size_t n = end - begin;
+    size_t idx = g.get_local_id().get(0);
+    if(idx == 0)
+      for(size_t i = 0; i < n; ++i)
+        for(size_t j = i + 1; j < n; ++j)
+          if(comp(begin[j], begin[i]))
+            std::swap(begin[i], begin[j]);
+  }
+};
+----
+
+==== Predefined Sorters
+
+`radix_order` is a `enum` that defines the sorting order when `radix_sorter` is used.
+Only ascending and descending orders are applicable.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+  enum radix_order {
+    ascending,
+    descending
+  };
+
+}
+----
+
+SYCL provides the following predefined classes:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+  template<typename Compare = std::less<>>
+  class default_sorter {
+  public:
+    default_sorter(Compare comp = Compare());
+
+    template<typename Group, typename Ptr>
+    void operator()(Group g, Ptr begin, Ptr end);
+
+    template<typename Group, typename T>
+    T operator()(Group g, T val);
+  };
+
+  template<typename T, radix_order Order = radix_order::ascending, unsigned int BitsPerPass = 4>
+  class radix_sorter {
+  public:
+    radix_sorter(const std::bitset<sizeof(T) * CHAR_BIT> mask =
+                 std::bitset<sizeof(T) * CHAR_BIT> (std::numeric_limits<unsigned long long>::max()));
+
+    template<typename Group, typename Ptr>
+    void operator()(Group g, Ptr begin, Ptr end);
+
+    template<typename Group, typename T>
+    T operator()(Group g, T val);
+  };
+
+}
+----
+
+Table. Description of predefined Sorters.
+|===
+|Sorter|Description
+
+|`template<typename Compare = std::less<>>
+default_sorter`
+|Use a default sorting method based on an implementation-defined heuristic
+using `Compare` as the binary comparison function object.
+
+|`template<typename T, typename Compare = std::less<T>, unsigned int BitsPerPass = 4>
+radix_sorter`
+|Use radix sort as a sorting method. `Order` specify the sorting order.
+Only arithmetic types as `T` can be passed to radix_sorter.
+`BitsPerPass` is a number of bits that values are split by.
+For example, if a sequence of `int32_t` is sorted using `BitsPerPass == 4` then one
+pass of the radix sort algorithm considers only 4 bits. The number of passes is `32/4=8`.
+|===
+
+Table. Constructors of the `default_sorter` class.
+|===
+|Constructor|Description
+
+|`default_sorter(Compare comp = Compare())`
+|Creates the `default_sorter` object using `comp`.
+|===
+
+Table. Member functions of the `default_sorter` class.
+|===
+|Member function|Description
+
+|`template<typename Group, typename Ptr>
+void operator()(Group g, Ptr begin, Ptr end)`
+|Implements a default sorting algorithm. It's callable by the `joint_sort` algorithm.
+
+_Complexity_: Let `N` be `end - begin`. `O(Nlog^2(N))` comparisons.

```

Comment:
```suggestion
_Complexity_: Let `N` be `end - begin`: stem:[O(N \times log_2(N))] comparisons.
```

You can get much better formatting for these mathematical formulas by using the "stem" syntax in Asciidoctor.  Same comment for all the compexity formulas below too.

You also need to add this line to the top of the file (e.g. after the `:lang: en` line):

```
:stem:
```

Reply:
Tried. GitHub shows it as `:\$O(N \times log_2(N))\$.`
As for me, the previous version looked better.

Reply:
It looks like the GitHub Asciidoc preview feature doesn't support the default "stem" feature.  You could check to see if it supports "latexmath".  To do this, change the `:stem:` line at the top of the file to this:

```
:stem: latexmath
```

and then change the equation to `latexmath:[O(N \times log_2(N))]`

However, based on this comment in GitHub, my guess is that this won't work either:

https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74#gistcomment-3632305

Reply:
Yes. Seems, it also doesn't work.

Reply:
I think it's important to use features that are supported in GitHub, so I guess we should avoid using the "stem" syntax.  Maybe saying `O(N*log_2(N))` would be a little clearer than `O(Nlog^2(N))`, but either is fine.

Reply:
Yes. I think `O(N*log_2(N))` is a better since `log_2` can be more familiar for users than `log^2`: `log_2` is used by Wolfram, Tex